src/core/services/theming/theming.js

```diff
@@ -146,6 +149,9 @@ function ThemingProvider($mdColorPalette) {
     alwaysWatchTheme: function(alwaysWatch) {
       alwaysWatchTheme = alwaysWatch;
     },
+    generateThemesOnDemand: function(onDemand) {
```

Comment:
`generateThemesOnDemand` is less ambiguous. `enableOnDemand` doesn't say which action is actually happening on-demand.